exercise-app/src/test/java/org/exercise/app/AppControllerTest.java

```diff
@@ -0,0 +1,88 @@
+package org.exercise.app;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.examples.attsw.exercise.core.controller.IEmployeeController;
+import com.examples.attsw.exercise.core.model.Employee;
+
+public class AppControllerTest {
+
+	private AppController appController;
+	private IEmployeeController employeeController;
+	private String allEmployees;
+
+	@Before
+	public void setUp() throws Exception {
+		allEmployees = "";
+		employeeController = mock(IEmployeeController.class);
+		appController = new AppController(employeeController);
+	}
+
+	@Test
+	public void testShowAllWhenThereAreNoEmployees() {
+		assertShowWhat("There are no Employees", "showAll", "");
+	}
+
+	@Test
+	public void testShowAllWhenThereIsOneEmployee() {
+		allEmployees = concatNewEmployee("1", "name1");
+		assertShowWhat(allEmployees, "showAll", "");
+	}
+
+	@Test
+	public void testShowAllWhenThereAreTwoEmployees() {
+		allEmployees = concatNewEmployee("1", "name1").concat(concatNewEmployee("2", "name2"));
+		assertShowWhat(allEmployees, "showAll", "");
+	}
+
+	@Test
+	public void testShowOneWhenThereAreNoEmployees() {
+		assertShowWhat("There are no Employees", "showOne", "1");
+	}
+
+	@Test
+	public void testShowOneWheneEmployeeDoesNotExists() {
+		allEmployees = concatNewEmployee("1", "name1");
+		when(employeeController.getEmployeeById("2")).thenReturn("");
+		assertShowWhat("There are no Employee with this id", "showOne", "2");
+	}
+
+	@Test
+	public void testShowOneWhenEmployeeExists() {
+		Employee employee = createNewEmployee("1", "name1");
```

Comment:
This test should not need to create Employee objects: the App itself is agnostic concerning objects, since it only communicates with the controller.
When you stub the controller you should not need to use Employee objects but just return strings; how the controller returns strings is not this SUT's business.